haystack/nodes/prompt/invocation_layer/hugging_face.py

```diff
@@ -219,27 +219,11 @@ def invoke(self, *args, **kwargs):
         stream = stream or stream_handler is not None
         if kwargs and "prompt" in kwargs:
             prompt = kwargs.pop("prompt")
+            # remove parameters that are not accepted by the transformers' pipeline
+            kwargs.pop("documents", None)
+            kwargs.pop("query", None)
```

Comment:
@faaany are they being popped because they cause the generation to fail?

Reply:
If so, it seems like we'll need a more generic method where we remove all prompt template parameters before passing them to pipeline generation...